pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go

```diff
@@ -433,36 +433,45 @@ type KubernetesSDConfig struct {
 // See https://prometheus.io/docs/prometheus/latest/configuration/configuration/#consul_sd_config
 // +k8s:openapi-gen=true
 type ConsulSDConfig struct {
-	// A valid string consisting of a hostname or IP followed by an optional port number.
+	// Consul server address. A valid string consisting of a hostname or IP followed by an optional port number.
 	// +kubebuilder:validation:MinLength=1
 	// +required
 	Server string `json:"server"`
+	// Prefix for URIs for when consul is behind an API gateway (reverse proxy).
+	// +kubebuilder:validation:MinLength=1
+	// +optional
+	PathPrefix *string `json:"pathPrefix,omitempty"`
 	// Consul ACL TokenRef, if not provided it will use the ACL from the local Consul Agent.
 	// +optional
 	TokenRef *corev1.SecretKeySelector `json:"tokenRef,omitempty"`
 	// Consul Datacenter name, if not provided it will use the local Consul Agent Datacenter.
+	// +kubebuilder:validation:MinLength=1
 	// +optional
 	Datacenter *string `json:"datacenter,omitempty"`
 	// Namespaces are only supported in Consul Enterprise.
+	// +kubebuilder:validation:MinLength=1
 	// +optional
 	Namespace *string `json:"namespace,omitempty"`
 	// Admin Partitions are only supported in Consul Enterprise.
+	// +kubebuilder:validation:MinLength=1
 	// +optional
 	Partition *string `json:"partition,omitempty"`
 	// HTTP Scheme default "http"
 	// +kubebuilder:validation:Enum=HTTP;HTTPS
 	// +optional
 	Scheme *string `json:"scheme,omitempty"`
 	// A list of services for which targets are retrieved. If omitted, all services are scraped.
-	// +listType:=atomic
+	// +listType:=set
 	// +optional
 	Services []string `json:"services,omitempty"`
 	// An optional list of tags used to filter nodes for a given service. Services must contain all tags in the list.
-	//+listType:=atomic
+	// +listType:=set
+	// +kubebuilder:validation:MinItems=1
```

Comment:
can you remove it for now and see if the tests pass?

Reply:
Not sure what you mean here. Should I remove the failing test, or the MinItems validation and change the testcase accordingly?

Reply:
sorry for being dense: my advice is to remove the MinItems validation (for now) and remove the test.

Reply:
The tests pass!